src/Shared/HashCodeCombiner/HashCodeCombiner.cs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) .NET Foundation. All rights reserved.
```

Comment:
Suggest moving all C# files that, like this one, are alone in a folder up into src/Shared/

Reply:
I don't see much benefit to doing that though. It's not a patter we have in aspnetcore anyway. Also, I suspect we'll be getting these types from dotnet/runtime in the future.

Reply:
Huh? There's already a handful of C# files in the src/Shared/ folder. The extra folders like src/Shared/**HashCodeCombiner**/ just make it harder to find things.

Reply:
There is also SecurityHelper and PropertyActivator that only contain a single file so it's not a pattern we follow strictly.

Reply:
I agree we're inconsistent and want to fix this. The more I think about navigating our source, the less I like one-member folders in the tree. So, not a suggestion: Don't make usability worse in this PR. Up to you whether you fix up the `SecurityHelper` and `PropertyActivator` locations at the same time. (If you don't handle those two, please file an issue that's in the Build project.)

Reply:
I'll file an issue to do it after the extensions consolidation. I don't want to add more complexity here.

Reply:
Filed https://github.com/dotnet/aspnetcore/issues/19080.

Reply:
(2:30 AM Saturday morning? Get some sleep 😴)